urlshortener/mainapp/views.py

```diff
@@ -1,6 +1,31 @@
-from django.shortcuts import render
+from django.shortcuts import get_object_or_404, render, redirect
 from django.http import HttpResponse
-
+from .models import Shortener
+from .forms import NewUrlForm
 # Create your views here.
 def home(req):
-    return HttpResponse("<h1>Hello! This is just checking we are working fine!</h1>")
\ No newline at end of file
+    return HttpResponse("<h1>Hello! This is just checking we are working fine!</h1>")
+
+def show(request, url):
+    url_object = Shortener.objects.filter(short_url=url)
+    if (len(url_object) > 0):
+        url_to_find = url_object[0].long_url
+        return redirect(url_to_find)
+    return HttpResponse('')
+
+def create(request):
+    if request.method == 'POST':
+        url = NewUrlForm(request.POST)
+        if url.is_valid():
```

Comment:
I'm not 100% on this without trying it. But you don't have an else after this if so I think an invalid form would maybe cause unexpected behaviour. I know you're validating on the client side though so in theory it shouldn't ever be invalid.